src/diffusers/models/transformers/transformer_z_image_omni.py

```diff
@@ -0,0 +1,1005 @@
+                        sig_padded_feats.append(sig_item_padded_feat)
+                        sig_noise_mask.extend([images_noise_mask[i][j]] * (sig_item_ori_len + sig_item_padding_len))
+                    else:
+                        sig_pad_dim = 1152
+                        sig_item_ori_len = 0
+                        sig_item_padding_len = SEQ_MULTI_OF
+                        sig_item_padding_pos_ids = (
+                        self.create_coordinate_grid(
+                                size=(1, 1, 1),
+                                # start=(cap_start_pos[j], 0, 0),
+                                start=(0, 0, 0),
+                                device=device,
+                            )
+                            .flatten(0, 2)
+                            .repeat(sig_item_padding_len, 1)
+                        )
+                        sig_padded_pos_ids.append(sig_item_padding_pos_ids)
+                        sig_pad_mask.append(torch.ones((sig_item_padding_len,), dtype=torch.bool, device=device))
+                        sig_padded_feats.append(torch.zeros((sig_item_padding_len, sig_pad_dim), dtype=dtype, device=device))
+                        sig_noise_mask.extend([images_noise_mask[i][j]] * sig_item_padding_len)
+
+                all_siglip_noise_mask.append(sig_noise_mask)
+                sig_padded_pos_ids = torch.cat(sig_padded_pos_ids, dim=0)
+                all_siglip_pos_ids.append(sig_padded_pos_ids)
+                sig_pad_mask = torch.cat(sig_pad_mask, dim=0)
+                all_siglip_pad_mask.append(sig_pad_mask)
+                all_siglip_padded_feats.append(torch.cat(sig_padded_feats, dim=0))
+                all_siglip_len.append(sig_len)
+                # print(f"> all_siglip_feats[i]: {all_siglip_feats[i].shape}", flush=True)
+
+        all_x_pos_offsets = []
+        for i in range(bsz):
+            start = sum(all_cap_len[i])
+            end = start + sum(all_x_len[i])
+            all_x_pos_offsets.append((start, end))
+            assert all_x_padded[i].shape[0] + all_cap_padded_feats[i].shape[0] == sum(all_cap_len[i]) + sum(all_x_len[i]), f"Batch item {i}: x length {all_x_padded[i].shape[0]} + cap length {all_cap_padded_feats[i].shape[0]} != sum(all_cap_len[i]) + sum(all_x_len[i]) {sum(all_cap_len[i]) + sum(all_x_len[i])}"
+
+        return (
+            all_x_padded,
+            all_cap_padded_feats,
+            all_siglip_padded_feats,
+            all_x_size,
+            all_x_pos_ids,
+            all_cap_pos_ids,
+            all_siglip_pos_ids,
+            all_x_pad_mask,
+            all_cap_pad_mask,
+            all_siglip_pad_mask,
+            all_x_pos_offsets,
+            all_x_noise_mask,
+            all_cap_noise_mask,
+            all_siglip_noise_mask,
+        )
+
+    def forward(
+        self,
+        x: List[torch.Tensor],
+        t,
+        cap_feats: List[List[torch.Tensor]],
+        cond_latents: List[List[torch.Tensor]],
```

Comment:
```suggestion
```
can we cancat with the latent inside the pipeline? like  
https://github.com/huggingface/diffusers/blob/55463f7ace24f0506a38c3971291e63d50bc989d/src/diffusers/pipelines/qwenimage/pipeline_qwenimage_edit.py#L812